clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/RequestBasedMetadata.java

```diff
@@ -260,19 +268,17 @@ private synchronized boolean updateCache(boolean onDemandRefresh) throws Interru
         updateCache(true);
       } else {
         // pass the error along if the on demand refresh also fails
+        clusterStats.recordVersionUpdateFailure();
         throw new VeniceClientException("Metadata fetch retry has failed", e.getCause());
       }
     }
-
-    return newVersion;
   }
 
   private void refresh() {
     try {
-      if (updateCache(false)) {
-        if (routingStrategy instanceof HelixScatterGatherRoutingStrategy) {
-          ((HelixScatterGatherRoutingStrategy) routingStrategy).updateHelixGroupInfo(helixGroupInfo);
-        }
+      updateCache(false);
+      if (routingStrategy instanceof HelixScatterGatherRoutingStrategy) {
```

Comment:
The code style can be improved in the future.
For example, we can always expose a function: `updateHelixGroupInfo` in the abstract class, and the default impl will do nothing and `HelixScatterGatherRoutingStrategy` will override it to refresh the local state.

Reply:
Good call, this is a small enough change that I can just do it in this PR. Will do it now.